### ISSUE TITLE
Option for target directory

### ISSUE THREAD

**Issue by andymeneely** [JamesIves/github-pages-deploy-action#80]
**Is your feature request related to a problem? Please describe.**

I have a lot of output files from my builds (~150 images) that I'd like to deploy to a subdirectory, e.g. `assets/cards/`. Currently, if I specify my output folder, it copies all 150 images to the root of the repository.

To work around this at the moment, I specify `FOLDER: .` and move my images in my source repo. But I'd rather not copy my entire repo over to the gh-pages target - just this folder.

**Describe the solution you'd like**

An option, say, `DEST_FOLDER` that specifies the destination on the ghpages branch. For backward compatibility, it could default to `.`

**Additional Comments**

This is a non-blocking issue for me, but it would be nice.

Love this action! Thank you for doing it.

**Comment by JamesIves** [comment]
I think this change would only require a minor adjustment to this line: https://github.com/JamesIves/github-pages-deploy-action/blob/dev/src/git.ts#L89-L98

I'm happy to add this later this week.

**Comment by JamesIves** [comment]
@andymeneely I still need to write some documentation, but I wanted to get this in your hands as soon as possible for you to test.

You can use the test version of the action with this functionality by referencing the following:

```
uses: JamesIves/github-pages-deploy-action@releases/v3-test
```

You can then add a parameter for `TARGET_FOLDER` like so.

```yml
- name: Build and Deploy Repo
   uses: JamesIves/github-pages-deploy-action@releases/v3-test
   with:
     ACCESS_TOKEN: ${{ secrets.ACCESS_TOKEN }}
     BRANCH: gh-pages
     FOLDER: build
     TARGET_FOLDER: mytest
```

When the action deploys it will push anything in `FOLDER` into `TARGET_FOLDER`. If the variable is not included then it will go straight into the root.

Let me know if that works!

**Comment by JamesIves** [comment]
Documentation is completed, I'll merge this Monday unless there's any objections.

**Comment by JamesIves** [comment]
Tagged and released as `3.0.5`. Available in `releases/v3`.

https://github.com/JamesIves/github-pages-deploy-action/releases/tag/3.0.5
